domain/ide/converter/converter.go

```diff
@@ -154,30 +156,243 @@ func ToPosition(p snyk.Position) sglsp.Position {
 	}
 }
 
-func ToDiagnostics(issues []snyk.Issue) []types.Diagnostic {
+func ToDiagnostics(issues []snyk.Issue, pr product.Product) []types.Diagnostic {
 	// In JSON, `nil` serializes to `null`, while an empty slice serializes to `[]`.
 	// Sending null instead of an empty array leads to stored diagnostics not being cleared.
 	// Do not prefer nil over an empty slice in this case. The next line ensures that even if issues is empty,
 	// the return value of this function will not be null.
-	diagnostics := []types.Diagnostic{}
 
-	for _, issue := range issues {
+	var issueListForDiagnostic []types.ScanIssue
+	// check product type
+	if pr == product.ProductInfrastructureAsCode {
+		issueListForDiagnostic = appendIacIssues(issueListForDiagnostic, issues)
+	} else if pr == product.ProductCode {
+		issueListForDiagnostic = appendCodeIssues(issueListForDiagnostic, issues)
+	} else if pr == product.ProductOpenSource {
+		issueListForDiagnostic = appendOssIssues(issueListForDiagnostic, issues)
+	}
+
+	var diagnostics []types.Diagnostic
+
+	for i, issue := range issues {
 		s := ""
 		if issue.IssueDescriptionURL != nil {
 			s = issue.IssueDescriptionURL.String()
 		}
-		diagnostics = append(diagnostics, types.Diagnostic{
+		diagnostic := types.Diagnostic{
 			Range:           ToRange(issue.Range),
 			Severity:        ToSeverity(issue.Severity),
 			Code:            issue.ID,
 			Source:          string(issue.Product),
 			Message:         issue.Message,
 			CodeDescription: types.CodeDescription{Href: types.Uri(s)},
-		})
+		}
+		// TODO: This is not future proof. Find a better way to match.
```

Comment:
the appendIaCIssues/Code/Oss functions should now be changed to enrich 1 issue instead of receiving a list. This will fix this TODO.